### PR TITLE
New kinds of policies

### PR DESCRIPTION
**Pull request by quarckster**
{{pytest: cfme/tests/control/test_basic.py -k test_policies_copy --long-running}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 5d003a4db171892c91cd1d2cea899c63c63d44d1:

`cfme/tests/control/test_basic.py`:
- :bangbang: Line 438:9: **E131** *continuation line unaligned for hanging indent*

Please, rectify these issues :smirk: .
*CFME QE Bot*

File: cfme/tests/control/test_basic.py [hunk 1]
```diff
@@ -55,14 +55,36 @@ def random_alert():
...
+        explorer.ContainerNodeCompliancePolicy,
+        explorer.ContainerImageCompliancePolicy,
+        explorer.HostControlPolicy,
+        explorer.VMControlPolicy,
+        explorer.ReplicatorControlPolicy,
+        explorer.PodControlPolicy,
+        explorer.ContainerNodeControlPolicy,
+        explorer.ContainerImageControlPolicy
+    ],
+    ids=[
```
Comment (mfalesni) [inline cfme/tests/control/test_basic.py]: Now it got quite long and I do believe the ids= can also take callables, so something like:
```python
ids=lambda policy_class: policy_class.__name__
```

should do this typing for you

File: cfme/tests/control/test_basic.py [hunk 2]
```diff
@@ -404,6 +426,16 @@ def test_container_image_compliance_policy_crud(soft_assert):
...
 @pytest.mark.tier(3)
+@pytest.mark.uncollectif(
+    lambda random_policy: random_policy in (explorer.ReplicatorCompliancePolicy,
+                                            explorer.PodCompliancePolicy,
+                                            explorer.ContainerNodeCompliancePolicy,
+                                            explorer.ContainerImageCompliancePolicy,
+                                            explorer.ReplicatorControlPolicy,
+                                            explorer.PodControlPolicy,
+                                            explorer.ContainerNodeControlPolicy,
+                                            explorer.ContainerImageControlPolicy) and
```
Comment (mfalesni) [inline cfme/tests/control/test_basic.py]: Pull this list out from this lambda so it lies in this module and make it a set. Name it eg. `CONTAINER_POLICIES` so it is obvious :)

**Comment by dajoRH** [comment]
Lint report for commit 76a3108dddc810c3e5060366c20ed10b4886773b:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/control/explorer.py [hunk 3]
```diff
@@ -511,21 +307,40 @@ def f(_=None):
 class _type_check_object(object):
```
Comment (mfalesni) [inline cfme/control/explorer.py]: You could do something like:
```python
class _type_check_object(object):
    TYPES = set()

    @classmethod
    def register_type(cls, type):
        cls.TYPES.add(type)
        return type

    def _is_assignable(self, what):
        for klass in _type_check_object.TYPES:
            if isinstance(self, klass):
                return True
        else:
            return False
```

and then:

```python
@_type_checkobject.register_type
class HostObject(_type_check_object):
    pass
```

**Comment by dajoRH** [comment]
Lint report for commit 00de3b0bb7c150b23ea7a2443494a5af9826d126:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 79d64e776e4a14ce8700fe0465543d01a63b98b4:

`cfme/control/explorer.py`:
- :heavy_exclamation_mark: Line 242:25: **W291** *trailing whitespace*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9214677d682b4f8ff8442a817c8d5aa062c2a66d:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 22eabe75d16055f92d1a9b8d3e6a844a3d160d50:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d931b003577f0a02c3edac23285a3052ae0f9b65:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 61a8c38fb7c6d895750e67321ade225fa7bb428e:

`cfme/control/explorer.py`:
- :bangbang: Line 236:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 238:13: **E121** *continuation line under-indented for hanging indent*
- :bangbang: Line 240:17: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 242:13: **E121** *continuation line under-indented for hanging indent*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c838c5da396e9aeef6e7151c1396f53be1dfa4b3:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6a8b1c817e4e1c9ca642e9b5a064ffba014895eb:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 41ea49ad10c5fef90a55f1e91cb3e27421279029:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 782bead2211170280c91e98608efbdec510e1620:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3957974107cbcd621279258104abc04386b358b1:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 437e8cc121a497cfbda6a474e34bb724f524ca9d:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 86150d6e78efa72bf72df2a30f93e5e3edeb2eaa:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9c2c1f8e2832ecd66b88476d2211d54fd2b976c0:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request merged**